tests/libnet/netattachdef.go

```diff
@@ -0,0 +1,153 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package libnet
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"maps"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+const (
+	// ResourceNameAnnotation represents a resource name that is associated with the network.
+	// It could be found on NetworkAttachmentDefinition objects.
+	ResourceNameAnnotation = "k8s.v1.cni.cncf.io/resourceName"
+)
+
+type pluginConfOption func(map[string]interface{})
+
+func NewBridgeNetAttachDef(name, bridgeName string, opts ...pluginConfOption) *nadv1.NetworkAttachmentDefinition {
+	const cniPluginType = "bridge"
+
+	pluginConfig := NewNetPluginConfig(
+		cniPluginType,
+		map[string]interface{}{"bridge": bridgeName},
+	)
+	for _, f := range opts {
+		f(pluginConfig)
+	}
+	return NewNetAttachDef(name, NewNetConfig(name, pluginConfig))
+}
+
+func NewSriovNetAttachDef(name string, vlanID int, opts ...pluginConfOption) *nadv1.NetworkAttachmentDefinition {
+	const pluginType = "sriov"
+	pluginConf := map[string]interface{}{"vlan": vlanID}
+	for _, f := range opts {
+		f(pluginConf)
+	}
+	return NewNetAttachDef(name, NewNetConfig("sriov", NewNetPluginConfig(pluginType, pluginConf)))
+}
+
+func NewPasstNetAttachDef(name string) *nadv1.NetworkAttachmentDefinition {
+	const pluginType = "kubevirt-passt-binding"
+	return NewNetAttachDef(name, NewNetConfig(name, NewNetPluginConfig(pluginType, nil)))
+}
+
+func NewMacvtapNetAttachDef(name, lowerDevice string) *nadv1.NetworkAttachmentDefinition {
+	const pluginType = "macvtap"
+	netAttachDef := NewNetAttachDef(name, NewNetConfig(name, NewNetPluginConfig(pluginType, nil)))
+	netAttachDef.Annotations = map[string]string{ResourceNameAnnotation: "macvtap.network.kubevirt.io/" + lowerDevice}
+	return netAttachDef
+}
+
+func NewNetAttachDef(name, config string) *nadv1.NetworkAttachmentDefinition {
+	return &nadv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec:       nadv1.NetworkAttachmentDefinitionSpec{Config: config},
+	}
+}
+
+func NewNetConfig(name string, pluginsConfigs ...map[string]interface{}) string {
+	const cniVersion = "0.3.1"
```

Comment:
Unrelated question: Is there an issue preventing us from upgrading to the latest CNI standard version?